gtd-timer/Controllers/TasksController.cs

```diff
@@ -0,0 +1,237 @@
+            )
+        {
+            this.taskService = taskService;
+            this.userIdentityService = userIdentityService;
+        }
+
+
+        [HttpGet("[action]")]
+        public IActionResult GetAllTasks()
+        {
+            try
+            {
+                var allTasks = taskService.GetAllTasks();
+                if (allTasks != null)
+                {
+                    return Ok(allTasks);
+                }
+
+                return NoContent();
+            }
+            catch (Exception e)
+            {
+                return BadRequest(e);
+            }
+        }
+
+        [HttpGet("[action]")]
+        public IActionResult GetAllActiveTasks()
+        {
+            try
+            {
+                var allTasks = taskService.GetAllActiveTasks();
+                if (allTasks != null)
+                {
+                    return Ok(allTasks);
+                }
+
+                return NoContent();
+            }
+            catch (Exception e)
+            {
+                return BadRequest(e);
+            }
+        }
+
+        [HttpGet("[action]")]
+        public IActionResult GetAllArchivedTasks()
+        {
+            try
+            {
+                var allTasks = taskService.GetAllArchivedTasks();
+                if (allTasks != null)
+                {
+                    return Ok(allTasks);
+                }
+
+                return NoContent();
+            }
+            catch (Exception e)
+            {
+                return BadRequest(e);
+            }
+        }
+
+        [HttpGet("[action]")]
+        public IActionResult GetTaskById(int taskId)
+        {
+            TaskDTO task = taskService.GetTaskById(taskId);
+
+            return Ok(task);
+        }
+
+        [HttpGet("[action]")]
+        public IActionResult GetAllTasksByUserId()
+        {
+            try
+            {
+                var userId = userIdentityService.GetUserId();
+                var tasks = taskService.GetAllTasksByUserId(userId);
+                if (tasks != null)
+                {
+                    return Ok(tasks);
+                }
+
+                return NoContent();
+            }
+            catch (Exception e)
+            {
+                return BadRequest(e);
+            }
+        }
+
+        [HttpGet("[action]")]
+        public IActionResult GetAllActiveTasksByUserId()
+        {
+            try
+            {
+                var userId = userIdentityService.GetUserId();
+                var tasks = taskService.GetAllActiveTasksByUserId(userId);
+                if (tasks != null)
+                {
+                    return Ok(tasks);
+                }
+
+                return NoContent();
+            }
+            catch (Exception e)
+            {
+                return BadRequest(e);
+            }
+        }
+
+        [HttpGet("[action]")]
+        public IActionResult GetAllArchivedTasksByUserId()
+        {
+            try
```

Comment:
all those try catch are unnecessary - our exception handling midleware should do the trick